hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -73,6 +71,16 @@
   private static final Logger LOG = LogManager.getLogger(HoodieActiveTimeline.class);
   protected HoodieTableMetaClient metaClient;
   private static AtomicReference<String> lastInstantTime = new AtomicReference<>(String.valueOf(Integer.MIN_VALUE));
+  private static ThreadLocal<SimpleDateFormat> commitFormatHolder = new ThreadLocal<SimpleDateFormat>() {
```

Comment:
Have you considered using `DateTimeFormatter` which is thread-safe and immutable? :)

Reply:
+1 @prashantwason . Introducing a thread local for this scenario seems like an overkill?

Reply:
@ssdong DateFimeFormatter is a good suggestion. But its format() method does not take a Date() but takes a TemporalAccessory (i.e.e LocalDateTime, etc). So the amount of changes would still be significant.

@vinothchandar Yeah, we can work without thread local


What do you both think of this idea?

1. Change COMMIT_FORMATER to be DateFimeFormatter but make it private
2. Introduce a new method   
       HoodieActiveTimeline.format(time_in_msec)

Reply:
yes. we can make a `LocalDate` from `Date` easily. (https://www.baeldung.com/java-date-to-localdate-and-localdatetime) 

+1 on the new static method. In fact, we can even have a separate class `HoodieInstantTimeGenerator` and keep an singleton instance in HoodieActiveTimeline. We can then add methods to `HoodieInstantTimeGenerator` as needed.

Reply:
Created a new class.